homeassistant/components/input_boolean/reproduce_state.py

```diff
@@ -0,0 +1,50 @@
+"""Repdroduce a light state."""
+import asyncio
+import logging
+from typing import Iterable, Optional
+
+from homeassistant.const import (
+    SERVICE_TURN_OFF,
+    SERVICE_TURN_ON,
+    STATE_ON,
+    ATTR_ENTITY_ID,
+)
+from homeassistant.core import Context, State
+from homeassistant.helpers.typing import HomeAssistantType
+
+from . import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def _async_reproduce_states(
+    hass: HomeAssistantType, state: State, context: Optional[Context] = None
+) -> None:
+    """Reproduce input boolean states."""
+    cur_state = hass.states.get(state.entity_id)
+
+    if cur_state is None:
+        _LOGGER.warning("Unable to find entity %s", state.entity_id)
+        return
+
+    if cur_state.state == state.state:
+        return
+
+    service = SERVICE_TURN_ON if state.state == STATE_ON else SERVICE_TURN_OFF
```

Comment:
If only set an attribute, but not state. State will be None. This will cause turn off, when imho error should be reported.